etcdserver/corrupt.go

```diff
@@ -24,6 +24,69 @@ import (
 	"github.com/coreos/etcd/pkg/types"
 )
 
+func (s *EtcdServer) checkHashKVInit() {
+	// TODO: separate configuration for initial hash check?
+	if s.Cfg.CorruptCheckTime < 3*time.Second {
+		return
+	}
+
+	select {
+	case <-s.readych: // sync with s.publish
+	case <-s.stopping:
+		return
+	}
+	mbs := s.cluster.Members()
+	n := len(mbs)
+	if n < 3 {
+		plog.Infof("no need to check corruption on %s (%d members)", s.ID().String(), n)
+		return
+	}
+	plog.Infof("corruption checking on %s (%d members)", s.ID().String(), n)
+
+	h, rev, _, err := s.kv.HashByRev(0)
+	if err != nil {
+		plog.Fatalf("failed to hash kv store (%v)", err)
+	}
+
+	mismatch := 0
+	for _, m := range mbs {
+		if m.ID == s.ID() {
+			continue
+		}
+
+		cli, cerr := clientv3.New(clientv3.Config{Endpoints: m.PeerURLs})
+		if cerr != nil {
+			continue
+		}
+
+		var resp *clientv3.HashKVResponse
+		for _, c := range cli.Endpoints() {
+			ctx, cancel := context.WithTimeout(context.Background(), s.Cfg.ReqTimeout())
+			resp, cerr = cli.HashKV(ctx, c, rev)
+			cancel()
+			if cerr == nil {
+				break
+			}
+		}
+		cli.Close()
+
+		if resp == nil && cerr != nil {
+			plog.Fatal(cerr)
```

Comment:
if all its peers are dead, then this peer cannot start. the cluster cannot be bootstrapped after a full shutdown?